worker/migrations/migrate-accounts-to-organisation.go

```diff
@@ -0,0 +1,282 @@
+	"github.com/cozy/cozy-stack/model/bitwarden"
+	"github.com/cozy/cozy-stack/model/bitwarden/settings"
+	"github.com/cozy/cozy-stack/model/instance"
+	"github.com/cozy/cozy-stack/model/job"
+	"github.com/cozy/cozy-stack/pkg/consts"
+	"github.com/cozy/cozy-stack/pkg/couchdb"
+	"github.com/cozy/cozy-stack/pkg/crypto"
+	"github.com/cozy/cozy-stack/pkg/metadata"
+	"github.com/cozy/cozy-stack/web/data"
+	"github.com/sirupsen/logrus"
+
+	multierror "github.com/hashicorp/go-multierror"
+)
+
+type VaultReference struct {
+	ID       string `json:"_id"`
+	Type     string `json:"_type"`
+	Protocol string `json:"_protocol"`
+}
+
+func isAdditionalField(fieldName string) bool {
+	return !(fieldName == "login" ||
+		fieldName == "password" ||
+		fieldName == "advancedFields")
+}
+
+// Builds a cipher from an io.cozy.account
+//
+// A raw JSONDoc is used to be able to access auth.fields
+func buildCipher(orgKey []byte, manifest *app.KonnManifest, account couchdb.JSONDoc, url string, logger *logrus.Entry) (*bitwarden.Cipher, error) {
+	logger.Infof("Building ciphers...")
+
+	auth, _ := account.M["auth"].(map[string]interface{})
+
+	username, _ := auth["login"].(string)
+	password, _ := auth["password"].(string)
+	email, _ := auth["email"].(string)
+
+	// Special case if the email field is used instead of login
+	if username == "" && email != "" {
+		username = email
+	}
+
+	key := orgKey[:32]
+	hmac := orgKey[32:]
+
+	ivURL := crypto.GenerateRandomBytes(16)
+	encURL, err := crypto.EncryptWithAES256HMAC(key, hmac, []byte(url), ivURL)
+	if err != nil {
+		return nil, err
+	}
+	u := bitwarden.LoginURI{URI: encURL, Match: nil}
+	uris := []bitwarden.LoginURI{u}
+
+	ivName := crypto.GenerateRandomBytes(16)
+	encName, err := crypto.EncryptWithAES256HMAC(key, hmac, []byte(manifest.Name), ivName)
+	if err != nil {
+		return nil, err
+	}
+
+	ivUsername := crypto.GenerateRandomBytes(16)
+	encUsername, err := crypto.EncryptWithAES256HMAC(key, hmac, []byte(username), ivUsername)
+	if err != nil {
+		return nil, err
+	}
+
+	ivPassword := crypto.GenerateRandomBytes(16)
+	encPassword, err := crypto.EncryptWithAES256HMAC(key, hmac, []byte(password), ivPassword)
+	if err != nil {
+		return nil, err
+	}
+
+	login := &bitwarden.LoginData{
+		Username: encUsername,
+		Password: encPassword,
+		URIs:     uris,
+	}
+
+	md := metadata.New()
+	md.DocTypeVersion = bitwarden.DocTypeVersion
+
+	bitwardenFields := make([]bitwarden.Field, 0)
+
+	for name, rawValue := range auth {
+		value, ok := rawValue.(string)
+		if !ok {
+			continue
+		}
+		if !isAdditionalField(name) {
+			continue
+		}
+
+		ivName := crypto.GenerateRandomBytes(16)
+		encName, err := crypto.EncryptWithAES256HMAC(key, hmac, []byte(name), ivName)
+		if err != nil {
+			return nil, err
+		}
+
+		ivValue := crypto.GenerateRandomBytes(16)
+		encValue, err := crypto.EncryptWithAES256HMAC(key, hmac, []byte(value), ivValue)
+		if err != nil {
+			return nil, err
+		}
+
+		logger.Infof("Adding field %s = %s", name, value)
```

Comment:
What is the purpose of this log? Can we remove it to avoid leaking encrypted values to logs?

Reply:
I thought I had removed this log, thanks.